plutus-core/plutus-ir/src/PlutusIR/Compiler/Provenance.hs

```diff
@@ -27,23 +27,14 @@ data Provenance a = Original a
                   | DatatypeComponent DatatypeComponent (Provenance a)
                   -- | Added for accumulating difference provenances when floating lets
                   | MultipleSources (S.Set (Provenance a))
-                  deriving stock (Show, Eq, Ord)
+                  deriving stock (Show, Eq, Ord, Foldable)
 
 instance Ord a => Semigroup (Provenance a) where
   MultipleSources ps1 <> MultipleSources ps2 = MultipleSources (ps1<>ps2)
   x <> MultipleSources ps2                   = MultipleSources (S.insert x ps2)
   MultipleSources ps1 <> x                   = MultipleSources (S.insert x ps1)
   x <> y                                     = MultipleSources (S.fromList [x,y])
```

Comment:
Completely unrelated:

this could be

```haskell
x <> y = MultipleSources (provenanceToSet x <> provenanceToSet y)
```

assuming `provenanceToSet` is defined somewhere.

(that has a slightly different semantics, but I guess we don't care)

Reply:
I like that, but perhaps in a separate commit.